src/web/components/SlotBodyEnd.tsx

```diff
@@ -9,11 +9,19 @@ const wrapperMargins = css`
 export const SlotBodyEnd = () => {
     const endpointUrl = 'https://contributions.guardianapis.com/epic';
 
-    const trackingParams = {};
+    const trackingParams = {
+        ophanPageId: window.guardian.config.ophan?.pageViewId,
```

Comment:
I'm not sure what's best here but this raises the question: should we pass in data via props or read it all from a global window object? If we want scalability I'd be inclined to lean on react and use props but I'm not sure what the goals are here?

Reply:
I think there's an ordering dependency here on ophan-tracker-js, which may or may not get loaded early enough for this. cc @oliverlloyd and @gtrufitt . Probably we want to load Ophan as a priority script as a few things likely rely on `pageViewId` being available early?

Reply:
The Ophan ID we see here is set as a global variable (inside window.guardian...) in the <head> on DCR, to ensure it's set as early as possible and before the app is rehydrated.

Reply:
Basically - the Ophan Page is generated and added to the global scope independently of any Ophan scripts - https://github.com/guardian/dotcom-rendering/blob/b5e01f6d2ec98122a0ca443793bde4aa145ae6d3/src/web/server/htmlTemplate.ts#L117 - with it being placed in the `<head>` it should be reliable enough.

Reply:
This all makes a lot more sense to me now. Thanks for clarifying. Happy to move forwards

Reply:
Update: turns out we set this in `htmlTemplate` so the ophan-tracker-js library doesn't set it on DCR (so no race-condition).